src/GraphQL.Tests/Subscription/SubscriptionTests.cs

```diff
@@ -0,0 +1,112 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Threading.Tasks;
+using GraphQL.Subscription;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace GraphQL.Tests.Subscription
+{
+    public class SubscriptionTests
+    {
+        [Fact]
+        public async Task Subscribe()
+        {
+            /* Given */
+            var addedMessage = new Message
+            {
+                Content = "test",
+                From = new MessageFrom()
+                {
+                    DisplayName = "test",
+                    Id = "1"
+                },
+                SentAt = DateTime.Now
+            };
+            var chat = new Chat();
+            var schema = new ChatSchema(chat);
+            var sut = new SubscriptionExecuter();
+
+            /* When */
+            var result = await sut.SubscribeAsync(new ExecutionOptions
+            {
+                Query = "subscription MessageAdded { messageAdded { from { id displayName } content sentAt } }",
+                Schema = schema
+            });
+
+            chat.AddMessage(addedMessage);
+
+            /* Then */
+            var stream = result.Streams.Values.FirstOrDefault();
+            var message = await stream.FirstOrDefaultAsync();
+            Assert.NotNull(message);
+            Assert.IsType<ExecutionResult>(message);
```

Comment:
Mind switching these assertions to Shouldly?  That is what the rest of the project is using.

Reply:
Of course.